src/TextFieldPage.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from 'react';
+import ComponentPage from './ComponentPage.js';
+import HeaderBar from './HeaderBar.js';
+import {MDCTextField} from '@material/textfield';
+import classnames from 'classnames';
+
+import './styles/TextFieldPage.scss';
+
+const getLabel = (disabled, dense) => {
+  if (disabled) {
+    return 'Disabled';
+  } else if (dense) {
+    return 'Dense';
+  }
+  return 'Standard';
+}
+
+const TextField = (props) => {
+  const {
+    textFieldId, box, outlined, fullwidth, textarea,
+    disabled, dense, leading, trailing, helperText,
+  } = props;
+  const classes = classnames('mdc-text-field', 'text-field', {
+    'mdc-text-field--box': box,
+    'mdc-text-field--outlined': outlined,
+    'mdc-text-field--fullwidth': fullwidth,
+    'mdc-text-field--textarea': textarea,
+    'mdc-text-field--disabled': disabled,
+    'mdc-text-field--dense': dense,
+    'mdc-text-field--with-leading-icon': leading,
+    'mdc-text-field--with-trailing-icon': trailing,
+  });
+
+  const showPlaceholder = fullwidth && !textarea;
```

Comment:
I dont think that you should have so much logic in our catalog pages. This showPlaceholder is difficult to understand.